src/access/wac.ts

```diff
@@ -235,3 +319,92 @@ export function getGroupAccessAll(
 ): Promise<AgentWacAccess | null> {
   return getActorAccessAll(resource, getGroupAccessAllWac, options);
 }
+
+async function setActorResourceAccess<T extends WithServerResourceInfo>(
+  resource: T,
+  actor: UrlString,
+  access: WacAccess,
+  readAccess: typeof getAgentAccessWac,
+  writeAccess: typeof setAgentResourceAccessWac,
+  options: Partial<typeof internal_defaultFetchOptions>
+): Promise<(T & WithResourceAcl) | null> {
+  if (!hasAccessibleAcl(resource)) {
+    return null;
+  }
+  const acl = await internal_fetchAcl(resource, options);
+  const resourceWithAcl = internal_setAcl(resource, acl);
+  let resourceAcl: AclDataset;
+  if (hasResourceAcl(resourceWithAcl)) {
+    // This is the simple case, where the Resource ACL we need to update already
+    // exists.
+    resourceAcl = internal_getResourceAcl(resourceWithAcl);
+  } else if (hasFallbackAcl(resourceWithAcl)) {
+    // In this case, the Resource ACL needs to be created first, and then updated.
+    resourceAcl = createAclFromFallbackAcl(resourceWithAcl);
+  } else {
+    return null;
+  }
+
+  const resourceWithOldAcl = internal_setResourceAcl(
+    resourceWithAcl,
+    resourceAcl
+  );
+  let wacAccess: AclAccess | undefined = undefined;
+  if (hasNoUndefinedAccessModes(access)) {
+    // If all the access modes in `access` are set, no need for the previous access.
+    wacAccess = universalAccessToAcl(access, undefined);
```

Comment:
You shouldn't need this if you update the overloads as above. That said, might it not be easier/more readable to just remove `hasNoUndefinedAccessModes` and its related types altogether, and just always run `readAccess` anyway?

```suggestion
    wacAccess = universalAccessToAcl(access);
```

Reply:
dbaafa53cb91f1e4d0b853f2bcbd36859739cda6